src/jit/codegenarmarch.cpp

```diff
@@ -655,8 +655,15 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 LclVarDsc* varDsc = &compiler->lvaTable[varNumInp];
 
                 assert(varDsc->lvType == TYP_STRUCT);
-                assert(varDsc->lvOnFrame);   // This struct also must live in the stack frame
-                assert(!varDsc->lvRegister); // And it can't live in a register (SIMD)
+                if (varDsc->lvPromoted)
+                {
+                    NYI_ARM("CodeGen::genPutArgStk - promoted struct");
+                }
+                else
```

Comment:
I don't think so. It's because the promoted struct is not on frame and may live in registers. 
If I changed the code like you proposed, the promoted struct is processed (by code from the future) and then, it also checks the promoted struct should be on frame and not live in registers.

But I know your point, so I'll find new way to keep different behavior for ARM64.